tasks/update-configuration.yml

```diff
@@ -20,14 +20,53 @@
 - name: "Set listen_address value"
   ansible.builtin.lineinfile:
     dest: "{{ gitlab_runner_config_path }}"
-    regexp: '^(\s*)listen_address =.*'
+    regexp: '^listen_address =.*'
     line: 'listen_address = "{{ gitlab_runner_listen_address }}"'
     state: present
     insertafter: '^\s*concurrent =.*'
   when: "not gitlab_runner_is_initial_dryrun"  # skip if run for the first time in check mode
   notify:
     - Restart GitLab-Runner
 
+- name: "Set session_server listen_address value"
+  ansible.builtin.lineinfile:
+    dest: "{{ gitlab_runner_config_path }}"
+    regexp: '^\[session_server\]\n(\s*)listen_address =.*'
+    line: '  listen_address = "{{ gitlab_runner_session_server_listen_address }}"'
+    state: present
+    insertafter: '^\[session_server\].*'
+  when:
+    - "not gitlab_runner_is_initial_dryrun"  # skip if run for the first time in check mode
+    - "gitlab_runner_enable_session_server"
+  notify:
+    - Restart GitLab-Runner
+
+- name: "Set session_server advertise_address value"
+  ansible.builtin.lineinfile:
+    dest: "{{ gitlab_runner_config_path }}"
+    regexp: '^(\s*)advertise_address =.*'
+    line: '  advertise_address = "{{ gitlab_runner_session_server_advertise_address }}"'
+    state: present
+    insertafter: '^\[session_server\]\n(\s*)listen_address =.*'
```

Comment:
@tobiashuste This does not seem to work as expected for some reason. The result is always: 
```toml
[session_server]
  listen_address = "0.0.0.0:8093"
  session_timeout = 1800
  advertise_address = "0.0.0.0:8093"
```

Reply:
Btw: Even when `gitlab_runner_enable_session_server` is set to `false` the runner always configures the following:
```toml
[session_server]
  session_timeout = 1800
```

I tried the suggested order `listen_address` :arrow_right: `session_timeout` :arrow_right: `advertise_address` but it doesn't make a difference.

Reply:
Now, it should be [fixed](https://github.com/hifis-net/ansible-role-gitlab-runner/compare/9e43bee776d02f4077f2a69736211c5107c14b31..2eaa240a8b7d99115c423746f939f928ae816b14).